sentry/src/main/java/io/sentry/protocol/Feedback.java

```diff
@@ -0,0 +1,233 @@
+package io.sentry.protocol;
+
+import io.sentry.ILogger;
+import io.sentry.JsonDeserializer;
+import io.sentry.JsonSerializable;
+import io.sentry.JsonUnknown;
+import io.sentry.ObjectReader;
+import io.sentry.ObjectWriter;
+import io.sentry.SentryLevel;
+import io.sentry.util.CollectionUtils;
+import io.sentry.util.Objects;
+import io.sentry.vendor.gson.stream.JsonToken;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// Specs can be found at https://develop.sentry.dev/sdk/data-model/envelope-items/#user-feedback
+
+public final class Feedback implements JsonUnknown, JsonSerializable {
+  public static final String TYPE = "feedback";
+
+  private @NotNull String message;
+  private @Nullable String contactEmail;
+  private @Nullable String name;
+  private @Nullable SentryId associatedEventId;
+  private @Nullable SentryId replayId;
+  private @Nullable String url;
+
+  private @Nullable Map<String, Object> unknown;
+
+  public Feedback(final @NotNull String message) {
+    this.message = message;
+  }
+
+  public Feedback(final @NotNull Feedback feedback) {
+    this.message = feedback.message;
+    this.contactEmail = feedback.contactEmail;
+    this.name = feedback.name;
+    this.associatedEventId = feedback.associatedEventId;
+    this.replayId = feedback.replayId;
+    this.url = feedback.url;
+    this.unknown = CollectionUtils.newConcurrentHashMap(feedback.unknown);
+  }
+
+  public @Nullable String getContactEmail() {
+    return contactEmail;
+  }
+
+  public void setContactEmail(final @Nullable String contactEmail) {
+    this.contactEmail = contactEmail;
+  }
+
+  public @Nullable String getName() {
+    return name;
+  }
+
+  public void setName(final @Nullable String name) {
+    this.name = name;
+  }
+
+  public @Nullable SentryId getAssociatedEventId() {
+    return associatedEventId;
+  }
+
+  public void setAssociatedEventId(final @NotNull SentryId associatedEventId) {
+    this.associatedEventId = associatedEventId;
+  }
+
+  public @Nullable SentryId getReplayId() {
+    return replayId;
+  }
+
+  public void setReplayId(final @NotNull SentryId replayId) {
+    this.replayId = replayId;
+  }
+
+  public @Nullable String getUrl() {
+    return url;
+  }
+
+  public void setUrl(final @Nullable String url) {
+    this.url = url;
+  }
+
+  public @NotNull String getMessage() {
+    return message;
+  }
+
+  public void setMessage(final @NotNull String message) {
+    this.message = message;
+  }
```

Comment:
according to develop there's a limit of `4096` characters for the message, is this something we wanna let relay handle?

Reply:
imo relay should just handle it so it's fine

iirc we're generally now relying on relay to truncate things vs doing it client side

Reply:
maybe worth checking

Reply:
nice catch!
Just checked and confirm the SDK should truncate it, as it's dropped otherwise